flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/TypeStrategies.java

```diff
@@ -419,6 +420,17 @@ public static TypeStrategy varyingString(TypeStrategy initialStrategy) {
                 return Optional.of(nullReplacementDataType);
             };
 
+    /** Type strategy specific for greatest and least. */
+    public static final TypeStrategy GREATEST_LEAST =
```

Comment:
I believe this is the same as `TypeStrategies.COMMON`, isn't it? In that case we don't need a separate strategy.